Submitted By: Robert Connolly <robert at linuxfromscratch dot org> (ashes)
Date: 2005-01-01
Initial Package Version: 0.9.7h
Upstream Status: Not submitted
Origin: None
Description: This patch enables the arc4random() function in OpenSSL, when
using the Glibc or uClibc arc4random patch.

Also see:
http://www.linuxfromscratch.org/hlfs/
http://www.linuxfromscratch.org/hints/downloads/files/entropy.txt

diff -Naur openssl-0.9.7h.orig/crypto/rand/rand_unix.c openssl-0.9.7h/crypto/rand/rand_unix.c
--- openssl-0.9.7h.orig/crypto/rand/rand_unix.c	2005-01-14 16:19:47.000000000 +0000
+++ openssl-0.9.7h/crypto/rand/rand_unix.c	2005-10-18 23:20:08.000000000 +0000
@@ -125,7 +125,7 @@
 #include <unistd.h>
 #include <time.h>
 
-#ifdef __OpenBSD__
+#if defined(__OpenBSD__) || defined(LIBC_HAS_ARC4RANDOM)
 int RAND_poll(void)
 {
 	u_int32_t rnd = 0, i;
